data/nat/choose.lean

```diff
@@ -81,3 +81,10 @@ end
 
 theorem fact_mul_fact_dvd_fact {n k : ℕ} (hk : k ≤ n) : fact k * fact (n - k) ∣ fact n :=
 by rw [←choose_mul_fact_mul_fact hk, mul_assoc]; exact dvd_mul_left _ _
+
+lemma nat.prime.dvd_choose {p k : ℕ} (hk : 0 < k) (hkp : k < p) (hp : prime p) : p ∣ choose p k :=
+(hp.dvd_mul.1 (show p ∣ choose p k * (fact k * fact (p - k)),
```

Comment:
Hm, I think you should linearize this proof. What about writing it as a `begin ... end` block using `rw` and `refine`?

Reply:
I made some changes, which should make it more readable hopefully